File: app/src/main/java/com/unis/gameplatfrom/api/HUDLoadDataSubscriber.java
package com.unis.gameplatfrom.api;

import android.content.Context;
import android.os.Handler;
import android.text.TextUtils;

import com.kaopiz.kprogresshud.KProgressHUD;

/**
 * Created by wulei on 2016/9/24.
 */

@SuppressWarnings("DefaultFileTemplate")
public abstract class HUDLoadDataSubscriber<T> extends SimpleDataSubscriber<T> {
    private KProgressHUD kProgressHUD;

    public HUDLoadDataSubscriber(Context ctx) {
        this(ctx, null);
    }

    public HUDLoadDataSubscriber(Context ctx, String msg) {
        kProgressHUD = KProgressHUD.create(ctx)
                .setStyle(KProgressHUD.Style.SPIN_INDETERMINATE)
                .setCancellable(true)
                .setAnimationSpeed(2)
                .setDimAmount(0.5f);

        if (!TextUtils.isEmpty(msg)) {
            kProgressHUD.setLabel(msg);
        }
        kProgressHUD.show();
    }

    @Override
    public void onComplete() {
        kProgressHUD.dismiss();
    }

    @Override
    public void exception(int code, String message) {
        kProgressHUD.setLabel(message);
        new Handler().postDelayed(() -> kProgressHUD.dismiss(), 1500);

    }

    public void dismissHud() {
        kProgressHUD.dismiss();
    }
}

File: kotlin/src/main/java/com/unis/kotlin/api/HUDLoadDataSubscriber.kt
package com.unis.kotlin.api

import android.content.Context
import android.os.Handler
import android.text.TextUtils
import com.kaopiz.kprogresshud.KProgressHUD
import org.greenrobot.eventbus.Subscribe

open class HUDLoadDataSubscriber<T> constructor() : SimpleDataSubscriber<T>(){

    private var kProgressHUD: KProgressHUD? = null


    constructor(ctx: Context): this(ctx,null)

    constructor(ctx: Context, msg: String?) : this() {
        kProgressHUD = KProgressHUD.create(ctx)
                .setStyle(KProgressHUD.Style.SPIN_INDETERMINATE)
                .setCancellable(true)
                .setAnimationSpeed(2)
                .setDimAmount(0.5f)

        if (!TextUtils.isEmpty(msg)) {
            kProgressHUD!!.setLabel(msg)
        }
        kProgressHUD!!.show()
    }

    override fun onComplete() {
        kProgressHUD!!.dismiss()
    }

    override fun exception(code: Int, message: String) {
        kProgressHUD!!.setLabel(message)
        Handler().postDelayed({ kProgressHUD!!.dismiss() }, 1500)

    }

    fun dismissHud() {
        kProgressHUD!!.dismiss()
    }

}